pkg/monitortests/network/disruptionserviceloadbalancer/monitortest.go

```diff
@@ -275,16 +275,18 @@ func (w *availability) WriteContentToStorage(ctx context.Context, storageDir, ti
 }
 
 func (w *availability) namespaceDeleted(ctx context.Context) (bool, error) {
-	_, err := w.kubeClient.CoreV1().Namespaces().Get(ctx, w.namespaceName, metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		return true, nil
-	}
+	for retry := 0; retry < 6; retry++ {
+		_, err := w.kubeClient.CoreV1().Namespaces().Get(ctx, w.namespaceName, metav1.GetOptions{})
```

Comment:
I think being in the loop here will just hammer it 6 times in a row without a pause in between and then exit.  It is the PollUntilContextTimeout that waits in between the calls.
https://github.com/openshift/origin/blob/e07b8e3e77ae250299f9bbb6bcd135ef2f6408d9/pkg/monitortests/network/disruptionserviceloadbalancer/monitortest.go#L304

Reply:
Well, I missed the sleep that you have but still not sure this is the best way.

Reply:
Ok, so if the error is nil we return false, nil

If the error is not nil and not `IsNotFound` we try up to 6 times with a 5 second sleep in between.  But we still always return false, nil.  So really when we just try a few more times in the same call with a smaller interval when we see an non `IsNotFound` but ultimately keep going until we return true or timeout..

Reply:
Are you wanting to quit polling after 6 consecutive failures?  Maybe preserve the error and return it at the end.
so `var err error` outside the loop and return `false,err` at the end of the function.  Should only return there after the 6 tries and no `IsNotFound` or nil error. 

https://github.com/openshift/origin/blob/e07b8e3e77ae250299f9bbb6bcd135ef2f6408d9/pkg/monitortests/network/disruptionserviceloadbalancer/monitortest.go#L290

Reply:
yes, that was my intention originally. I've updated it now. look ok?